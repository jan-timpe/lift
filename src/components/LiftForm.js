import React, { Component } from 'react'


export default class LiftForm extends Component {

  constructor(props) {
    super(props)
    this.state = {
      liftName: ''
    }
  }


  render() {
    return (
      <div class="LiftForm">

        <form onSubmit={ (event) => {
          event.preventDefault()
          this.props.addLift({ name: this.state.liftName })
          this.setState({ liftName: '' })
        }}>
          <div className="input-wrapper">
            <label htmlFor="lift_name">New lift:</label>
            <input type="text" placeholder="e.g., Squat"
              value={ this.state.liftName }
              onChange={ (evt) => {
              this.setState({ liftName: evt.target.value });
              }}
            />
          </div>
          <div className="input-wrapper">
            <input type="submit" value="Create" className="button"/>
          </div>
        </form>
      
      </div>
    )
  }


}




