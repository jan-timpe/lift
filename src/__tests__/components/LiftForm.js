import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LiftForm from '../../components/LiftForm'


describe(`LiftForm`, () => {

  // snapshot
  it(`should render`, () => {
    const mockAddLift = jest.fn()
    const wrapper = shallow(<LiftForm addLift={mockAddLift}/>)
    const component = wrapper
    expect(toJson(component)).toMatchSnapshot()
  })


  // form
  it(`should call addLift on submit`, () => {
    const mockAddLift = jest.fn()
    const preventDefault = jest.fn()
    const wrapper = shallow(<LiftForm addLift={mockAddLift}/>)
    const component = wrapper

    component.find('form').simulate('submit', { preventDefault })
    expect(mockAddLift.mock.calls.length).toEqual(1)
  })

})