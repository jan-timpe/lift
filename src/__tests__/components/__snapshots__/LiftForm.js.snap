// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiftForm should render 1`] = `
<div
  class="LiftForm"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="input-wrapper"
    >
      <label
        htmlFor="lift_name"
      >
        New lift:
      </label>
      <input
        onChange={[Function]}
        placeholder="e.g., Squat"
        type="text"
        value=""
      />
    </div>
    <div
      className="input-wrapper"
    >
      <input
        className="button"
        type="submit"
        value="Create"
      />
    </div>
  </form>
</div>
`;
